### PR TITLE
[gui] Detailed error message when GGUI is not available

### PR DESCRIPTION
**Pull request by feisuzhu**
Related issue = #5090

### CODE REVIEW

File: python/taichi/ui/utils.py [hunk 1]
```diff
@@ -64,8 +67,74 @@ def vec_to_euler(v):
...
+
+            raise GGUINotAvailableException(
+                "GGUI is not available since you have installed a restricted version of taichi."
+            )
+
+    except GGUINotAvailableException:
+        raise
+
+    except Exception:
+        pass
```
Comment (ailzhang) [inline python/taichi/ui/utils.py:138]: OOC: when do we expect to hit this? ;)

File: python/taichi/ui/utils.py [hunk 2]
```diff
@@ -1,9 +1,12 @@
+import platform
 from math import acos, asin, cos, sin
 
 from taichi._lib import core as _ti_core
 from taichi.lang.impl import default_cfg
 from taichi.lang.matrix import Vector
 
+import taichi
```
Comment (ailzhang) [inline python/taichi/ui/utils.py:8]: OOC: is this import necessary? ;)

**Review by ailzhang — COMMENTED** [review]
Nice work!

File: python/taichi/ui/utils.py (hunk 1 above)
Comment (feisuzhu) [reply python/taichi/ui/utils.py:138]: When the code is buggy :)
This is a defending mechanism, if the code fails, just fallback to the most general situation, or user will see some confusing errors.

File: python/taichi/ui/utils.py (hunk 2 above)
Comment (feisuzhu) [reply python/taichi/ui/utils.py:8]: Will move it closer to where it's used.

File: python/taichi/ui/utils.py (hunk 1 above)
Comment (ailzhang) [reply python/taichi/ui/utils.py:137]: Aha sg!

File: python/taichi/ui/utils.py [hunk 3]
```diff
@@ -64,8 +65,75 @@ def vec_to_euler(v):
...
     """Checks if the `GGUI` environment is available.
     """
-    if not _ti_core.GGUI_AVAILABLE:
-        raise Exception("GGUI Not Available")
+    if _ti_core.GGUI_AVAILABLE:
+        return
+
+    try:
+        # Try identifing the reason
+        import taichi
```
Comment (ailzhang) [inline python/taichi/ui/utils.py:106]: Btw this might not pass the pylint check: https://docs.taichi-lang.org/docs/contributor_guide#enforce-code-style

**Comment by qiao-bo** [comment]
PR title check fails at unrecognized tag, seems like we don't have `ui` tag. Change to `gui` and rebase to trigger the build. @feisuzhu btw, don't forget the CLA ;/

**Comment by feisuzhu** [comment]
> PR title check fails at unrecognized tag, seems like we don't have `ui` tag. Change to `gui` and rebase to trigger the build. @feisuzhu btw, don't forget the CLA ;/

done

**Review by ailzhang — APPROVED** [review]
Thanks!

**Pull request merged**